contracts/v0.8/utils/Errors.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.18;
+
+library Errors {
+    error NonZeroExitCode(int256 exit_code, string description);
+
+    // Exit codes
+    int256 constant USR_ILLEGAL_ARGUMENT = 16;
+    int256 constant USR_NOT_FOUND = 17;
+    int256 constant USR_FORBIDDEN = 18;
+    int256 constant USR_INSUFFICIENT_FUNDS = 19;
+    int256 constant USR_ILLEGAL_STATE = 20;
+    int256 constant USR_SERIALIZATION = 21;
+    int256 constant USR_UNHANDLED_MESSAGE = 22;
+    int256 constant USR_UNSPECIFIED = 23;
+    int256 constant USR_ASSERTION_FAILED = 24;
+    int256 constant START_FOR_ACTOR_SPECIFIC_EXIT_CODES = 32;
+
+    function revertWithDescriptionIfExitCodeIsNotZero(int256 exit_code) internal pure {
+        if (exit_code == 0) return;
+
+        string memory description;
+
+        if (exit_code == USR_ILLEGAL_ARGUMENT) description = "USR_ILLEGAL_ARGUMENT: invalid message parameters";
+        if (exit_code == USR_NOT_FOUND) description = "USR_NOT_FOUND: message referenced something that doesn't exist";
+        if (exit_code == USR_FORBIDDEN) description = "USR_FORBIDDEN: operation forbidden";
+        if (exit_code == USR_INSUFFICIENT_FUNDS) description = "USR_INSUFFICIENT_FUNDS: insufficient funds for operation";
```

Comment:
I would just say `USR_INSUFFICIENT_FUNDS` instead of additional `i`nsufficient funds for operation` which doesn't provide any more debug help, especially if it's only used in tests.